specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/ManagedEnvironmentsDaprSubscriptions.json

```diff
@@ -0,0 +1,266 @@
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the Dapr subscription.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DaprSubscription"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Dapr subscription with default route only": {
+            "$ref": "./examples/DaprSubscriptions_Get_BulkSubscribeAndScopes.json"
+          },
+          "Get Dapr subscription with bulk subscribe configuration and scopes": {
+            "$ref": "./examples/DaprSubscriptions_Get_DefaultRoute.json"
+          },
+          "GetDapr subscription with route rules and metadata": {
+            "$ref": "./examples/DaprSubscriptions_Get_RouteRulesAndMetadata.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "DaprSubscriptions"
+        ],
+        "summary": "Creates or updates a Dapr subscription.",
+        "description": "Creates or updates a Dapr subscription in a Managed Environment.",
+        "operationId": "DaprSubscriptions_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "description": "Name of the Managed Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the Dapr subscription.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "daprSubscriptionEnvelope",
+            "in": "body",
+            "description": "Configuration details of the Dapr subscription.",
+            "required": true,
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DaprSubscription"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```

Comment:
For sync PUT, the 201 is still required to indicate the resource is created, refs: https://github.com/Azure/azure-openapi-validator/blob/main/docs/put-response-codes.md